sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala

```diff
@@ -1763,6 +1763,8 @@ object TypeCoercionSuite {
     Seq(DoubleType, FloatType, DecimalType.SYSTEM_DEFAULT, DecimalType(10, 2))
   val numericTypes: Seq[DataType] = integralTypes ++ fractionalTypes
   val datetimeTypes: Seq[DataType] = Seq(DateType, TimestampType, TimestampNTZType)
+  val intervalTypes: Seq[DataType] = Seq(CalendarIntervalType, DayTimeIntervalType.DEFAULT,
+    YearMonthIntervalType.DEFAULT)
```

Comment:
nit: I am thinking you should use `defaultConcreteType` to access the DEFAULT instance.